server/src/test/java/org/candlepin/resource/test/ConsumerResourceTest.java

```diff
@@ -521,6 +527,51 @@ public void testNullPerson() {
     }
 
     @Test
+    public void testCreateConsumerShouldFailOnMaxLengthOfName() {
+        thrown.expect(BadRequestException.class);
+        int max = Consumer.MAX_LENGTH_OF_CONSUMER_NAME;
+        String m = String.format("Name of the consumer " +
+            "should be shorter than %d characters.", max);
+        thrown.expectMessage(m);
+
+        Consumer c = mock(Consumer.class);
+        Owner o = mock(Owner.class);
+        UserPrincipal up = mock(UserPrincipal.class);
+        OwnerCurator oc = mock(OwnerCurator.class);
+        ConsumerTypeCurator ctc = mock(ConsumerTypeCurator.class);
+        ConsumerType cType = new ConsumerType(ConsumerTypeEnum.SYSTEM);
+        ConsumerResource consumerResource = new ConsumerResource(
+            null, null, null, null, null, null,
+            null, i18n, null, null, null, null,
+            null, null, null, null, oc, null, null, null,
+            null, null, null, new CandlepinCommonTestConfig(), null, null, null,
+            null, null, null);
+
+        String ownerKey = "testOwner";
+        when(o.getKey()).thenReturn(ownerKey);
+        when(oc.lookupByKey(eq(ownerKey))).thenReturn(o);
+        when(c.getType()).thenReturn(cType);
+        when(c.getName()).thenReturn(generateNameLongerThan255());
+        when(ctc.lookupByLabel(eq("system"))).thenReturn(cType);
+        when(up.canAccess(eq(o), eq(SubResource.CONSUMERS), eq(Access.CREATE))).
+            thenReturn(true);
+
+        consumerResource.create(c, up, null, ownerKey, null);
+    }
+
+    private String generateNameLongerThan255() {
+        String name255 =
+            "qwert12345qwert12345qwert12345qwert12345qwert12345" +
```

Comment:
In the future, for values which are derived from a defined constant like this we should be generating a string rather than hard-coding it.